pwndbg/commands/__init__.py

```diff
@@ -23,10 +23,26 @@
 
 commands = []
 
+def list_current_commands():
+    gdb.execute('set pagination off')
```

Comment:
In theory this is already handled by https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/__init__.py#L130

But if the user changes it afterwards it is not `off` in here.

Still, if the user changes it, they might want to keep their settings as they were, so we should be careful when setting things outside of initialization.

We should either:
* extend GDB's Python api so it is possible to get `help all` through it (long term solution; harder to implement)
* have our own `with statement` that can set given setting and unset it (revert to previous value) afterwards

Reply:
As is, this code is run before https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/__init__.py#L130 but I completely agree with the general point. Your long term solutions are great but I will probably just go do something like:
1. Save current pagination state
2. Turn it off
3. Call "help all"
4. Restore previous state

And I can look into the setting context handler separately. Where do you think would be an appropriate place to place such code?

Reply:
Hmm... not sure about the place. I've created `pwndbg/gdbutils` some time ago with an idea to move and abstract out all gdb related APIs (so we can support e.g. lldb in the loooong future). Maybe there?